liblepton/scheme/lepton/ffi.scm.in

```diff
@@ -127,3 +172,27 @@
 (define (pointer->geda-object pointer)
   ;; Return #f if the pointer is wrong.
   (false-if-exception (pointer->scm (edascm_from_object pointer))))
+
+;;; This syntax rule is intended for use in toplevel 'define' or
+;;; 'let' forms in the functions where the check for wrong type of
+;;; OBJECT is necessary.  The rule checks the object and, if it is
+;;; not #<geda-object>, throws an error with the 'wrong-type-arg
+;;; key reporting the function name and position POS of the
+;;; OBJECT argument.  In short, the usage is as follows:
+;;;   (define (myfunc object)
+;;;     (define pointer (geda-object->pointer* object 1))
+;;;     (function-body))
+(define-syntax-rule (geda-object->pointer* object pos)
```

Comment:
How this perverted name `geda-object->pointer*`
reflects the purpose of the function?

Reply:
I just haven't found anything better yet.  Please suggest another name you would like better.  The purpose of the function is transforming `#<geda-object>` to a pointer with some checks.  Defining it as syntax makes it inlinable which is convenient to get the name of the procedure it is called in.